File: Algorithm/baekjoon/BJ1541.java
package baekjoon;

import java.util.*;

/**
 * Created by changmin on 2017. 12. 12..
 */
public class BJ1541 {
    public static int calMininum(String input){
        int startIndex = 0;
        boolean minusFlag = false;
        int value = 0;
        int output = 0;

        for(int i = 0; i < input.length(); ++i){
            char token = input.charAt(i);

            if(token == '-' || token == '+'){
                value = Integer.parseInt(input.substring(startIndex, i));

                if(minusFlag)
                    output -= value;
                else
                    output += value;

                if(token == '-')
                    minusFlag = true;

                startIndex = i + 1;
            }
        }
        value = Integer.parseInt(input.substring(startIndex));

        if(minusFlag)
            output -= value;
        else
            output += value;

        return output;
    }

    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);

        String input = scanner.nextLine();

        int output = calMininum(input);

        System.out.println(output);
    }
}

File: Algorithm/baekjoon/swift/BJ1541.swift
func bracket(input: String) -> Int{
    var output = 0
    var start = input.startIndex
    var minus = false
    var index = 0
    for token in input{
        let i = input.index(input.startIndex, offsetBy: index)
        if token == "+" || token == "-" {
            let value = Int(input[start..<i])!
            start = input.index(after: i)
            
            if minus{
                output -= value
            }else{
                output += value
            }
            
            if token == "-"{
                minus = true
            }
        }
        
        index += 1
    }
    
    let value = Int(input[start...])!
    if minus{
        output -= value
    }else{
        output += value
    }
    
    return output
}
var input = readLine()
if let input = input{
    let output = bracket(input: input)
    print(output)
}
